WireAPI/Tests/WireAPITests/TeamsAPITests.swift

```diff
@@ -0,0 +1,446 @@
+        await assertAPIError(TeamsAPIError.teamNotFound) {
+            // When
+            _ = try await sut.getTeam(for: Team.ID())
+        }
+    }
+
+    func testGetTeamRolesForID_SuccessResponse_200_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            jsonResponse: """
+            {
+                "conversation_roles": [
+                    {
+                        "conversation_role": "admin",
+                        "actions": [
+                            "add_conversation_member",
+                            "remove_conversation_member"
+                        ]
+                    }
+                ]
+            }
+            """
+        )
+
+        let sut = TeamsAPIV0(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getTeamRoles(for: Team.ID())
+
+        // Then
+        XCTAssertEqual(
+            result,
+            [
+                ConversationRole(
+                    name: "admin",
+                    actions: [
+                        .addConversationMember,
+                        .removeConversationMember
+                    ]
+                )
+            ]
+        )
+    }
+
+    func testGetTeamRolesForID_FailureResponse_NoTeamMember_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 403, errorLabel: "no-team-member")
+        let sut = TeamsAPIV0(httpClient: httpClient)
+
+        // Then
+        await assertAPIError(TeamsAPIError.selfUserIsNotTeamMember) {
+            // When
+            _ = try await sut.getTeamRoles(for: Team.ID())
+        }
+    }
+
+    func testGetTeamRolesForID_FailureResponse_TeamNotFound_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 404, errorLabel: "")
+        let sut = TeamsAPIV0(httpClient: httpClient)
+
+        // Then
+        await assertAPIError(TeamsAPIError.teamNotFound) {
+            // When
+            _ = try await sut.getTeamRoles(for: Team.ID())
+        }
+    }
+
+    func testGetMembers_SuccessResponse_200_V0() async throws {
+        // Given
+        let userID = UUID()
+        let creatorID = UUID()
+        let creationDate = Date()
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            jsonResponse: """
+            {
+                "hasMore": true,
+                "members": [
+                    {
+                        "user": "\(userID.transportString())",
+                        "permissions": {
+                            "copy": 123,
+                            "self": 456
+                        },
+                        "created_by": "\(creatorID.transportString())",
+                        "created_at": "\(ISO8601DateFormatter.default.string(from: creationDate))",
+                        "legalhold_status": "pending"
+                    }
+                ]
+            }
+            """
+        )
```

Comment:
To keep readability in the tests one can move larger json to the bottom of the file and use a static var:
```swift
private enum JSON {
     static let testGetMembersV0 = """
     """
}
```

Or even create a new json file :)

Reply:
Yeah I was thinking about having these canned payloads in separate files that we load.

Reply:
done f67733603197f30a86f22b090ad13eef0ac96925